File: DayOne/Python/level_one.py
import math

def even_and_odd(numbers):
    even = []
    odd = []
    for number in numbers:
        if number % 2 == 0:
            even.append(number)
        else:
            odd.append(number)
            
    return [even, odd]

def is_palindrome(numbers):
    if numbers == numbers[::-1]:
        return True
        
    return False


def get_perfect_squares(numbers):
    result = []
    for number in numbers:
        if number >= 0:
            square_root = math.sqrt(number)
            if square_root * square_root == number:
                result.append(number)
    return result
    
def get_perfect_squares_to_negative(numbers):
    result = []
    for number in numbers:
        if number >= 0:
            square_root = math.sqrt(number)
            if square_root * square_root != number:
                result.append(-1)
            else:
                result.append(number)    
            
    return result

File: DayOne/Java/LevelOne.java
import java.util.Arrays;

public class LevelOne{
//LevelOne
    public static int[][] evenAndOddNumbers(int[] numbers){
        int oddCount = 0;
        int evenCount = 0;

        for (int number : numbers) {
            if (number % 2 == 0) {
                evenCount++;
            } else {
                oddCount++;
            }
}
        int[] even = new int[evenCount];
        int[] odd = new int[oddCount];

        int odds = 0;
        int evens = 0;

        for (int number : numbers) {
            if (number % 2 == 0) {
                even[evens++] = number;
            } else {
                odd[odds++] = number;
            }
        }
        return new int[][]{odd, even}; 
    }
    
    public static boolean isPalindrome(int[] numbers){
        int left  = 0;
        int right = numbers.length - 1;

        while (left < right) {
            if (numbers[left] != numbers[right])
            return false;
            left++;
            right--;
        }
        return true;
    }
    
    public static int[] perfectSquare(int[] numbers){
        int count = 0;
        for (int number : numbers) {
            int squareRoot = (int)Math.sqrt(number);
            if (squareRoot * squareRoot == number){
            count++;
            }
        }
        int[] result = new int[count];
        int index = 0;

        for (int number : numbers) {
            int squareRoot = (int)Math.sqrt(number);
            if (squareRoot * squareRoot == number){
            result[index++] = number;
        }
        }
        return result;
    }
}


